### PR TITLE
feat: get schema enum values by custom path

### PR DESCRIPTION
**Pull request by jessicamcinchak**
See https://github.com/theopensystemslab/planx-new/pull/6935 for context / implementation !

### CODE REVIEW

File: src/utils/digitalPlanningSchema.ts [hunk 1]
```diff
@@ -53,3 +53,38 @@ export function getValidSchemaDictionary(
...
+    if (basePath) {
+      const values =
+        // Default most common enum path
+        basePath["enum"] ||
+        // Handle one-level deep of nested properties
+        basePath["properties"]?.[nestedProperty]?.["enum"] ||
+        // Declaration uses an enumerated "value", free text "description"
+        basePath["properties"]?.["value"]?.["enum"] ||
+        // EPC, TitleNumber are discriminated unions and we want "known" variant
+        basePath["properties"]?.["known"]?.["enum"];
```
Comment (jessicamcinchak) [inline src/utils/digitalPlanningSchema.ts:74]: There's future edge cases we could add here in future, but this seems to capture _most_ values we want to access !

**Review by DafyddLlyr — APPROVED** [review]
Fab - appreciate the tests here 👌
